packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,422 @@
+    else {
+      return
+    }
+
+    let pendingTransactions = getPaymentQueueHandler().getUnfinishedTransactions()
+
+    for transaction in pendingTransactions {
+      // If the user cancels the purchase dialog we won't have a transactionIdentifier.
+      // So if it is null AND a transaction in the pendingTransactions list has
+      // also a null transactionIdentifier we check for equal product identifiers.
+      if transaction.transactionIdentifier == transactionIdentifier
+        || (transactionIdentifier == ""
+          && transaction.transactionIdentifier == nil
+          && transaction.payment.productIdentifier == productIdentifier)
+      {
+        getPaymentQueueHandler().finish(transaction)
+      }
+    }
+  }
+
+  public func restoreTransactionsApplicationUserName(
+    _ applicationUserName: String?, error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    getPaymentQueueHandler().restoreTransactions(applicationUserName)
+  }
+
+  public func presentCodeRedemptionSheetWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      getPaymentQueueHandler().presentCodeRedemptionSheet()
+    #endif
+  }
+
+  public func retrieveReceiptDataWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) -> String? {
+    var flutterError: FlutterError? = nil
+    if let receiptData = receiptManager.retrieveReceiptWithError(&flutterError) {
+      return receiptData
+    } else {
+      error.pointee = flutterError
+      return nil
+    }
+  }
+
+  public func refreshReceiptReceiptProperties(
+    _ receiptProperties: [String: Any]?, completion: @escaping (FlutterError?) -> Void
+  ) {
+    let properties = receiptProperties?.compactMapValues { $0 } ?? [:]
+    let request = getRefreshReceiptRequest(properties: properties.isEmpty ? nil : properties)
+    let handler = handlerFactory(request)
+    requestHandlers.insert(handler)
+    handler.startProductRequest { [weak self] response, error in
+      if let error = error {
+        let requestError = FlutterError(
+          code: "storekit_refreshreceiptrequest_platform_error",
+          message: error.localizedDescription,
+          details: error.localizedDescription)
+        completion(requestError)
+        return
+      }
+      completion(nil)
+      self?.requestHandlers.remove(handler)
+    }
+  }
+
+  public func startObservingPaymentQueueWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    getPaymentQueueHandler().startObservingPaymentQueue()
+  }
+
+  public func stopObservingPaymentQueueWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    getPaymentQueueHandler().stopObservingPaymentQueue()
+  }
+
+  public func registerPaymentQueueDelegateWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      if #available(iOS 13.0, *) {
+        let messenger = registrar?.messenger()
```

Comment:
nit: `guard let messenger = registrar?.messenger()`. 
It's better to deal with nullability up front, so that it's non-null for the rest of code